### PR TITLE
Integrate with encrypt

### PR DESCRIPTION
**Pull request by paraacha**
Module integrated with Encrypt module so to handle generation/removal of Encryption Profiles. Test Encrypt encryption method is being used at the moment because Real AES module is under development.

### CODE REVIEW

**Comment by nerdstein** [comment]
Looks good to me. @colans ?

File: pubkey_encrypt.install [hunk 1]
```diff
@@ -63,4 +63,20 @@ function pubkey_encrypt_install() {
...
+  $pubkey_encrypt_manager->initializeEncryptionProfiles();
+}
+
+/**
+ * Implements hook_uninstall().
+ */
+function pubkey_encrypt_uninstall() {
+  // Delete Encryption Profiles for all Role keys.
+  $keys = \Drupal::service('key.repository')
+    ->getKeysByProvider('pubkey_encrypt');
```
Comment (colans) [inline pubkey_encrypt.install:76]: Can probably keep this to one line if it's <= 80 characters and there's only one method call (no chaining).

File: pubkey_encrypt.module [hunk 2]
```diff
@@ -69,18 +69,37 @@ function pubkey_encrypt_form_submit($form, \Drupal\Core\Form\FormStateInterface
...
+  // Role keys so to give him complete control over them.
+  if ($account->hasPermission('administer permissions')) {
+    $pubkey_encrypt_manager->updateAllRoleKeys();
+  }
+  // Otherwise update the relevant Role keys so to give him access to those.
+  else {
+    foreach ($account->getRoles(TRUE) as $role) {
+      $role = \Drupal\user\Entity\Role::load($role);
+      $pubkey_encrypt_manager
+        ->updateRoleKey($role);
```
Comment (colans) [inline pubkey_encrypt.module:87]: Can probably keep this to one line if it's <= 80 characters and there's only one method call (no chaining).

File: src/PubkeyEncryptManager.php [hunk 3]
```diff
@@ -243,4 +244,42 @@ public function deleteRoleKey(Role $role) {
       ->delete();
   }
 
+  /*
+   * Initialize Encryption Profiles upon module installation.
+   */
+  public function initializeEncryptionProfiles() {
+    // Load all Role keys.
+    $keys = \Drupal::service('key.repository')
+      ->getKeysByProvider('pubkey_encrypt');
```
Comment (colans) [inline src/PubkeyEncryptManager.php:253]: Can probably keep this to one line if it's <= 80 characters and there's only one method call (no chaining).

File: src/PubkeyEncryptManager.php [hunk 4]
```diff
@@ -243,4 +244,42 @@ public function deleteRoleKey(Role $role) {
...
+
+    foreach ($keys as $key){
+      $this->generateEncryptionProfile($key);
+    }
+  }
+
+  /*
+   * Generate an Encryption profile for a Role key.
+   */
+  public function generateEncryptionProfile(Key $key) {
```
Comment (colans) [inline src/PubkeyEncryptManager.php:263]: `@param` is missing from the docblock above.  If this is the same as the parent, then `@inheritdoc` can be used.

File: src/PubkeyEncryptManager.php [hunk 5]
```diff
@@ -243,4 +244,42 @@ public function deleteRoleKey(Role $role) {
...
+    $this->entityTypeManager
+      ->getStorage('encryption_profile')
+      ->create($values)
+      ->save();
+  }
+
+  /*
+   * Remove the Encryption Profile for a Role key.
+   */
+  public function removeEncryptionProfile(Key $key) {
```
Comment (colans) [inline src/PubkeyEncryptManager.php:278]: `@param` is missing from the docblock above.  If this is the same as the parent, then `@inheritdoc` can be used.

**Pull request merged**